lib/generator/frameThroughputDctGenerator.py

```diff
@@ -190,30 +193,77 @@ def crop_images_based_on_samplefiles(self, input_data):
 
         return input_image_list
 
-    def generate_result(self, input_data, input_global_result):
+    def generate_result(self, input_data):
         """
 
         @param input_data:
         @return:
         """
 
-        compare_result = {}
+        self.compare_result = {}
 
         input_image_list = self.crop_images_based_on_samplefiles(input_data)
-        compare_result['merged_crop_image_list'] = input_image_list
+        self.compare_result['merged_crop_image_list'] = input_image_list
 
         # compare images
-        compare_setting = {'default_fps': input_data['index_config']['video-recording-fps'], 'event_points': self.BROWSER_VISUAL_EVENT_POINTS,
+        compare_setting = {'default_fps': self.index_config['video-recording-fps'],
+                           'event_points': self.BROWSER_VISUAL_EVENT_POINTS,
                            'generator_name': self.__class__.__name__,
                            'skip_status_bar_fraction': self.SKIP_STATUS_BAR_FRACTION,
                            'exec_timestamp_list': input_data['exec_timestamp_list'],
-                           'threshold': input_data['index_config']['compare-threshold'],
-                           'search_margin': input_data['index_config']['search-margin']}
-        compare_result['running_time_result'] = compare_with_sample_image_multi_process(input_data['sample_result'],
-                                                                                        input_image_list,
-                                                                                        compare_setting)
+                           'threshold': self.index_config.get('compare-threshold', 0.0003),
+                           'search_margin': self.index_config.get('search-margin', 10)}
+        self.compare_result['running_time_result'] = compare_with_sample_image_multi_process(
+            input_data['sample_result'],
+            input_image_list,
+            compare_setting)
         # get frame throughput values
-        compare_result.update(
-            self.get_frame_throughput(compare_result['running_time_result'], compare_result['merged_crop_image_list']))
+        if self.compare_result is not None:
```

Comment:
This line will be a logic implicit error, `self.compare_result` will always not equal to `None` even it's only equal to `{}`.
I guess here should to check the running time result exists or not to decide following calculation is needed or not, right?